From a85656d09079b5006d4ad4984a871254ba7c988f Mon Sep 17 00:00:00 2001
From: Joseph Chen <chenjh@rock-chips.com>
Date: Mon, 14 Aug 2017 15:27:33 +0800
Subject: [PATCH] power: rk818-battery: fix full charge sm_linek error

on the situation that: charger is connected and fuel gauge gives
state 'CC or CV', but actually battery is discharging from 100%
to 99%. At this moment we may treat this as normal full charging
by mistake, so we must add current judgement to avoid that.

Change-Id: I7b6692c17dc053fd96bd42df05c3f7763728bba2
Signed-off-by: Joseph Chen <chenjh@rock-chips.com>
---
 drivers/power/rk818_battery.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/power/rk818_battery.c b/drivers/power/rk818_battery.c
index caa4e23a8a8e..9ad6b5854450 100644
--- a/drivers/power/rk818_battery.c
+++ b/drivers/power/rk818_battery.c
@@ -2159,7 +2159,8 @@ static void rk818_bat_smooth_algorithm(struct rk818_battery *di)
 	di->remain_cap = rk818_bat_get_coulomb_cap(di);
 
 	/* full charge: slow down */
-	if ((di->dsoc == 99) && (di->chrg_status == CC_OR_CV)) {
+	if ((di->dsoc == 99) && (di->chrg_status == CC_OR_CV) &&
+	    (di->current_avg > 0)) {
 		di->sm_linek = FULL_CHRG_K;
 	/* terminal charge, slow down */
 	} else if ((di->current_avg >= TERM_CHRG_CURR) &&
-- 
2.35.3

